File: src/main/java/fi/finlit/edith/ui/services/UserInfoValueEncoder.java
/**
 * 
 */
package fi.finlit.edith.ui.services;

import org.apache.tapestry5.ValueEncoder;

import fi.finlit.edith.dto.UserInfo;

public class UserInfoValueEncoder implements ValueEncoder<UserInfo> {

    @Override
    public String toClient(UserInfo value) {
        return value.getUsername();
    }

    @Override
    public UserInfo toValue(String clientValue) {
        return new UserInfo(clientValue);
    }

}

File: src/main/scala/fi/finlit/edith/ui/services/UserInfoValueEncoder.scala
package fi.finlit.edith.ui.services

import org.apache.tapestry5.ValueEncoder
import fi.finlit.edith.dto.UserInfo
//remove if not needed
import scala.collection.JavaConversions._

class UserInfoValueEncoder extends ValueEncoder[UserInfo] {

  override def toClient(value: UserInfo): String = value.getUsername

  override def toValue(clientValue: String): UserInfo = new UserInfo(clientValue)
}
